src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -1016,7 +1019,41 @@ def _merge_signature(extracted, signature_overrides):
         return signature
 
     @staticmethod
-    def _infer_signature(
+    def _infer_signature(entry: Union[Callable, FlexFlow, Flow, Prompty], include_primitive_output: bool = False):
+        if isinstance(entry, Prompty):
+            from promptflow._sdk.schemas._flow import ALLOWED_TYPES
+            from promptflow.contracts.tool import ValueType
+            from promptflow.core._model_configuration import PromptyModelConfiguration
+
+            flow_meta = {"inputs": entry._data.get("inputs", {})}
+            if "outputs" in entry._data:
+                flow_meta["outputs"] = entry._data.get("outputs")
+            elif include_primitive_output:
+                flow_meta["outputs"] = {"output": {"type": "string"}}
+            init_dict = {}
+            for field in fields(PromptyModelConfiguration):
+                filed_type = type(field.type).__name__
+                init_dict[field.name] = {"type": filed_type if filed_type in ALLOWED_TYPES else ValueType.OBJECT.value}
+                if field.default != MISSING:
+                    init_dict[field.name]["default"] = field.default
+            flow_meta["init"] = init_dict
+            format_signature_type(flow_meta)
+        elif isinstance(entry, FlexFlow):
+            entry_func = entry_string_to_callable(entry.entry_file, entry.entry)
+            flow_meta, _, _ = FlowOperations._infer_signature_flex_flow(
+                entry=entry_func, language=entry.language, include_primitive_output=include_primitive_output
```

Comment:
for flex flow, include_primitive_output should be true here to cover flow with string output? CC: @elliotzh 
![image](https://github.com/microsoft/promptflow/assets/26239730/d2f4a8aa-8445-4d9f-8174-ce7b5898897b)